staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -45,15 +45,6 @@ import (
 // An object state is either "absent" or present with a
 // ResourceVersion and other appropriate content.
 //
-// A SharedInformer gets object states from apiservers using a
-// sequence of LIST and WATCH operations.  Through this sequence the
-// apiservers provide a sequence of "collection states" to the
-// informer, where each collection state defines the state of every
-// object of the collection.  No promise --- beyond what is implied by
-// other remarks here --- is made about how one informer's sequence of
-// collection states relates to a different informer's sequence of
-// collection states.
-//
```

Comment:
This paragraph makes sense even if "collection states" do not mirror what was stored by the API servers (as we now know).

I know you still want to remove it because it defines "collection states" and "collection states" are only referenced in a paragraph that gets removed by this PR too; IMO it should be kept, with an addition, because:

(1) It still contains useful information, i.e., a rough idea on how informers' caches are populated (LIST and WATCH) and the fact that there are no cross-informer ordering guarantees. I would explicitly add that a collection state that is seen by an informer did not necessarily exist in the API servers, or, equivalently, that the order in which objects (dis)appear in the informer's cache is not necessarily the order in which they (dis)appeared in the API servers.

(2) IMO the other paragraph that gets removed, i.e., the one referencing the concept of "collection states", should be kept (more details in my review comment on that paragraph).

Reply:
Since defining "collection states" is no longer needed, how about something along the lines of:

"A SharedInformer gets object states from API servers using a sequence of LIST and WATCH operations, and modifies its local cache to mirror the received object states. The order in which the states for a single object ID appear and disappear in the cache reflects the order in which those states appeared and disappeared in the authoritative state. The order in which the states for objects with different IDs appear and disappear in the cache does NOT necessarily reflect the order in which those states appeared and disappeared in the authoritative state. No promise --- beyond what is implied by other remarks here --- is made about how the order in which object states appear in one informer's cache relates to the order in which the same object states appear in another informer's cache". 

Also, assuming the aformentioned wording is actually used, IMO it would be better to swap its location with that of the paragraph that follows it, since the first mention to informers' local cache is in the latter.

Reply:
OK, I have drafted such a paragraph.  Having earlier defined "absent" as a state, we do not have to talk about disappearing.